src/main/java/beam/analysis/physsim/PhyssimCalcLinkSpeedDistributionStats.java

```diff
@@ -104,25 +103,22 @@ private CategoryDataset generateLinkEfficienciesDataSet(Map<Double, Integer> gen
 
     /**
      * Helper method that writes the final data to a CSV file
-     * @param dataMatrix the input data required to generate the charts
+     * @param data the input data required to generate the charts
      * @param outputFilePath path to the CSV file
      * @param heading header string for the CSV file
      */
-    private void writeCSV(double[][] dataMatrix,String outputFilePath,String heading) {
-        try {
-            BufferedWriter bw = new BufferedWriter(new FileWriter(outputFilePath));
+    private void writeCSV(double[] data, String outputFilePath,String heading) {
+        try(BufferedWriter bw = new BufferedWriter(new FileWriter(outputFilePath))) {
             String completeHeading = heading + ",numberOfLinks\n";
             bw.write(completeHeading);
-            double[] data = dataMatrix[0];
-            IntStream.range(0,data.length)
+            IntStream.range(0, data.length)
                     .forEach( i -> {
                         try {
                             bw.write(i + "," + data[i] + "\n");
                         } catch (IOException e) {
                             e.printStackTrace();
```

Comment:
It was like this before, but could you, please, add logging in all the places where `e.printStackTrace` is used?, e.g use 
```java
private final Logger log = LoggerFactory.getLogger(PhyssimCalcLinkSpeedDistributionStats.class);
```
to log the exception so we will be able to find it in the logs.

Reply:
Ok, I fixed it in part 2. I suggest merge it then I can create a new one PR.